src/sentry/integrations/jira_server/integration.py

```diff
@@ -1097,6 +1099,127 @@ def create_issue(self, data, **kwargs):
         # Immediately fetch and return the created issue.
         return self.get_issue(issue_key)
 
+    def _get_user_by_external_actor(
+        self,
+        client: JiraServerClient,
+        external_issue_key: str,
+        user: RpcUser,
+        logging_context: dict[str, Any],
+    ) -> list[Any] | None:
+        external_actors = ExternalActor.objects.filter(
+            organization_id=self.organization.id,
+            integration_id=self.model.id,
+            provider=ExternalProviders.JIRA_SERVER.value,
+            user_id=user.id,
+        )
+
+        if len(external_actors) > 1:
+            logger.warning(
```

Comment:
I'm unsure how we want to handle this case. Theoretically, a user can define multiple personal mappings. For now I'm just erroring, but we could theoretically loop through all of the Jira Server external actors for the user, looking for the first match.

Reply:
They don't 😅 The current logic is just to assume the username _is_ the email.

Agreed on not wanting to support this case with a 1:many mapping of sentry:jira users, but I'm not sure we have a constraint protecting this on the `ExternalActor` class at the moment. I'll explore this in the UI follow-up PR.